tests/app/Infrastructure/Controller/PostBuyCoinControllerTest.php

```diff
@@ -4,70 +4,89 @@
 
 use App\Domain\Coin;
 use App\Domain\DataSources\CoinDataSource;
-use App\Domain\DataSources\WalletDataSource;
-use App\Domain\Wallet;
+use Illuminate\Support\Facades\Cache;
 use Mockery;
+use Symfony\Component\HttpFoundation\Response;
 use Tests\TestCase;
 
 class PostBuyCoinControllerTest extends TestCase
 {
     private CoinDataSource $coinDataSource;
-    private WalletDataSource $walletDataSource;
 
     protected function setUp(): void
     {
         parent::setUp();
         $this->coinDataSource = Mockery::mock(CoinDataSource::class);
-        $this->walletDataSource = Mockery::mock(WalletDataSource::class);
         $this->app->bind(CoinDataSource::class, function () {
             return $this->coinDataSource;
         });
-        $this->app->bind(WalletDataSource::class, function () {
-            return $this->walletDataSource;
-        });
     }
 
     /**
      * @test
      */
-    public function ifCoinIdNotFoundThrowsError()
+    public function doesNotBuyCoinWhenCoinDoesNotExist()
     {
         $this->coinDataSource
             ->expects("findById")
             ->with("coin_id_value", "1")
             ->andReturn(null);
 
-        $response = $this->post('api/coin/buy', ["coin_id" => "coin_id_value",
-                                                      "wallet_id" => "wallet_id_value",
-                                                      "amount_usd" => 1]);
+        $response = $this->post(
+            'api/coin/buy',
+            [
+                "coin_id" => "coin_id_value",
+                "wallet_id" => "wallet_id_value",
+                "amount_usd" => 1
+            ]
+        );
 
-        $response->assertNotFound();
-        $response->assertExactJson(['description' => 'A coin with the specified ID was not found.']);
+        $response->assertStatus(Response::HTTP_NOT_FOUND);
+        $response->assertExactJson(
+            [
+                'description' => 'A coin with the specified ID was not found.'
+            ]
+        );
     }
 
     /**
      * @test
      */
-    public function ifBadRequestThrowsError()
+    public function doesNotBuyCoinWhenWalletDoesNotExist()
     {
         $this->coinDataSource
             ->expects("findById")
             ->with("coin_id_value", "1")
-            ->times(0)
-            ->andReturn(null);
+            ->andReturn(new Coin(
```

Comment:
Don't you think it is better to define the currency to a variable and return the variable here?